### PR TITLE
New samples with readme.

### PR DESCRIPTION
**Pull request by anthonyvercolano**
# Description of the pull request

### CODE REVIEW

File: provisioning/service/samples/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "azure-iot-provisioning-samples",
+  "version": "0.0.1",
+  "description": "Azure IoT SDK - Provisioning Service Client samples",
+  "author": "Microsoft Corporation",
+  "license": "MIT",
+  "main": "iotprovisioningservice.js",
+  "typings": "iotprovisioningservice.d.ts",
```
Comment (pierreca) [inline provisioning/service/samples/package.json]: probably not the right values

File: provisioning/service/samples/readme.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Samples for the Azure IoT Device Provisioning Service SDK for Node.js
+
+This file can be found in http://github.com/azure/azure-iot-sdk-node/provisioning/service/samples
+
+This folder contains simple samples showing how to use the various features of the Microsoft Azure IoT Device Provisioning Service from an application running JavaScript code.
+
+## List of samples
+
+* Create a simple IndividualEnrollment object.
+   *  **simple_IndividualEnrollment_create.js**
```
Comment (pierreca) [inline provisioning/service/samples/readme.md]: >**simple_IndividualEnrollment_create.js** [](start = 6, length = 41)

nit: these could easily be turned into links which would be slightly nicer.

File: provisioning/service/samples/readme.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+* Create queries for IndividualEnrollments, EnrollmentGroups and DeviceRegistrationStates that belong to an EnrollmentGroup.
+   *  **query_sample.js**
+* Perform BulkEnrollmentOperations to do various CRUD operations.
+   *  **bulkupdate_sample.js**
+
+
+## How to run the samples
+
+In order to run the device samples you will first need the following prerequisites:
+* Node.js v0.10 or above on your target device. (Check out [Nodejs.org](https://nodejs.org/) for more info)
```
Comment (pierreca) [inline provisioning/service/samples/readme.md]: >v0.10 [](start = 10, length = 5)

v4. our dependencies are dropping support for older versions and these aren't supported.

File: provisioning/service/samples/readme.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+* **package.json**
+* **__sample_file.js__** (where **__sample_file.js__** is one of the files listed above and available in this folder)
+
+Place the files in the folder of your choice on the target machine/device then go through the following steps:
+
+* From a shell or Node.js command prompt, navigate to the folder where you placed the sample files. For creating an IndividualEnrollment do:
+
+```
+$ npm install
+$ node sample_IndividualEnrollment.js "<the connection string for the Device Provisioning instance enclosed in quotes>" "<an endorsement key in quotes>"
```
Comment (pierreca) [inline provisioning/service/samples/readme.md]: >sample_IndividualEnrollment [](start = 7, length = 27)

missing _create ?

File: provisioning/service/samples/readme.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+<<<<<<<<<<<<<<<<<<<<<<<<<<<<<Link to how to create a signing certificate>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+
+* For creating an EnrollmentGroup do:
+
+```
+$ npm install
+$ node sample_EnrollmentGroup.js "<the connection string for the Device Provisioning instance enclosed in quotes>" "<The name of a file containing a pem or cer representation of a signing certificate>"
```
Comment (pierreca) [inline provisioning/service/samples/readme.md]: >sample_EnrollmentGroup [](start = 7, length = 22)

same ? missing _create?

File: provisioning/service/samples/readme.md (hunk 4 above)
Comment (pierreca) [reply provisioning/service/samples/readme.md]: also, weird casing mix in the file name is weird

---
In reply to: [150136510](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150136510) [](ancestors = 150136510)

Comment (pierreca) [reply provisioning/service/samples/readme.md]: (also if you decide not to change the name, i think it should be "simple" not "sample") but really this prefix doesn't add much value, does it?

---
In reply to: [150136587](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150136587) [](ancestors = 150136587,150136510)

File: provisioning/service/samples/package.json (hunk 1 above)
Comment (anthonyvercolano) [reply provisioning/service/samples/package.json]: Just get rid of the main and the typings?

---
In reply to: [150136067](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150136067) [](ancestors = 150136067)

File: provisioning/service/samples/readme.md (hunk 3 above)
Comment (anthonyvercolano) [reply provisioning/service/samples/readme.md]: Fixed.

---
In reply to: [150136312](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150136312) [](ancestors = 150136312)

File: provisioning/service/samples/readme.md (hunk 4 above)
Comment (anthonyvercolano) [reply provisioning/service/samples/readme.md]: See, the updated file names.

---
In reply to: [150136695](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150136695) [](ancestors = 150136695,150136587,150136510)

File: provisioning/service/samples/readme.md (hunk 5 above)
Comment (anthonyvercolano) [reply provisioning/service/samples/readme.md]: Done.

---
In reply to: [150136537](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150136537) [](ancestors = 150136537)

File: provisioning/service/samples/readme.md (hunk 2 above)
Comment (anthonyvercolano) [reply provisioning/service/samples/readme.md]: Done.

---
In reply to: [150136233](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150136233) [](ancestors = 150136233)

File: provisioning/service/samples/readme.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+   *  [run_bulk_operation.js][run-bulk-operation]
+
+
+## How to run the samples
+
+In order to run the device samples you will first need the following prerequisites:
+* Node.js v4 or above on your target device. (Check out [Nodejs.org](https://nodejs.org/) for more info)
+* [Setup Azure IoT Device Provisioning ][lnk-setup-iot-provisioning] Stop before executing the the Create and provision a simulated device section.  The samples below will perform these steps.
+
+<<<<<<<<<<<<<<<<<<link to how to obtain an endorsment key!!!!>>>>>>>>>>>>>>>>>>
```
Comment (nberdy) [inline provisioning/service/samples/readme.md]: This should be how to run the simulated TPM and extract the EK from it, yes?

File: provisioning/service/samples/readme.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+Place the files in the folder of your choice on the target machine/device then go through the following steps:
+
+* From a shell or Node.js command prompt, navigate to the folder where you placed the sample files. For creating an IndividualEnrollment do:
+
+```
+$ npm install
+$ node create_individual_enrollment.js "<the connection string for the Device Provisioning instance enclosed in quotes>" "<an endorsement key in quotes>"
+```
+
+<<<<<<<<<<<<<<<<<<<<<<<<<<<<<Link to how to create a signing certificate>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
```
Comment (nberdy) [inline provisioning/service/samples/readme.md]: Dhanashri is working on this documentation

Comment (nberdy) [reply provisioning/service/samples/readme.md]: What does creating a signing cert have to do with the sample flow? Should it be nested under the "For creating an EnrollmentGroup do" section?

**Comment by tameraw** [comment]
> * Create/remove/update/list enrollments and enrollment groups in your provisioning service

Also, "query" enrollments

---
Refers to: provisioning/service/readme.md:20 in a6885c3. [](commit_id = a6885c39575ea43de2ed8eefbedc419a5a366bb4, deletion_comment = False)

File: provisioning/service/samples/readme.md (hunk 6 above)
Comment (anthonyvercolano) [reply provisioning/service/samples/readme.md]: Something like that.

---
In reply to: [150296512](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150296512) [](ancestors = 150296512)

File: provisioning/service/samples/readme.md (hunk 7 above)
Comment (anthonyvercolano) [reply provisioning/service/samples/readme.md]: Yay!

---
In reply to: [150296581](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150296581) [](ancestors = 150296581)

Comment (anthonyvercolano) [reply provisioning/service/samples/readme.md]: Correct.   Moved.

---
In reply to: [150297054](https://github.com/Azure/azure-iot-sdk-node/pull/142#discussion_r150297054) [](ancestors = 150297054)

**Comment by anthonyvercolano** [comment]
> * Create/remove/update/list enrollments and enrollment groups in your provisioning service

Changed line to read:

Create, update, delete, query, and get individual enrollments and enrollment groups in your provisioning service.  Additionally you can query, get and delete device registration states.

---
In reply to: [343539997](https://github.com/Azure/azure-iot-sdk-node/pull/142#issuecomment-343539997) [](ancestors = 343539997)

---
Refers to: provisioning/service/readme.md:20 in a6885c3. [](commit_id = a6885c39575ea43de2ed8eefbedc419a5a366bb4, deletion_comment = False)

**Pull request merged**